src/wallet/injected/wallet-injected-interface.ts

```diff
@@ -0,0 +1,196 @@
+
+    /** Returns accounId from local storage or '' if it is not present */
+    getAccountId: () => string;
+
+    /* TODO: should it return other type of info if transaction was rejected by the user? */
+    /** TODOD: Should we return error type if transaction was not sent? */
+    /** On excecution of this function injected wallet should check if it can use Function call key
+     * to sign transaction and do that wihtout any prompts if possible.
+     * If this transaction requires Full access key,  user should be prompted.
+     * On approval transaction should be signed and sent.
+     * */
+    requestSignTransactions: (
+        params: RequestSignTransactionsInjectedOptions
+    ) => Promise<Array<FinalExecutionOutcome>>;
+}
+
+/**
+ * These options will become a part of the newly created Function Call key.
+ * {
+    public_key: 'ed25519:<public key>',
+    access_key: {
+      nonce: <nonce>,
+      permission: {
+        FunctionCall: {
+          allowance: null,
+          receiver_id: '<contractId>',              <------ @param contracId goes here
+          method_names: [<array of method names>] <------ @param methodNames goes here
+        }
+      }
+    }
+  }
+ */
+export interface SignInOptions {
+    contractId?: string;
+    methodNames?: string[];
+}
+
+export interface RequestSignTransactionsInjectedOptions {
+    /** List of transactions to sign */
+    transactions: Transaction[];
+    /** Meta information Wallet will send back to the application */
+    meta?: string; // TODO: should we have this field?
+    /** callback to be excecuted after function excecution */
+    callback? // TODO: should we have this field?
+}
+
+/////////////////////////// This section needs to be deleted after dicussions /////////////////////////////////
+export interface SenderWallet {
+    requestSignIn: (params: RequestSignInParams) => Promise<InitResponse>;
+    isSignedIn: () => boolean;
+    signOut: () => Promise<SignOutResponse>;
+    getAccountId: () => string;
+    requestSignTransactions: (
+        params: RequestSignTransactionsParams
+    ) => Promise<Array<SignAndSendTransactionResponse>>;
+
+    /* There is no reason to have an interface to sign one transaction,
+     * when we have interface for signing multiple transactions.
+     */
+    signAndSendTransaction: (
+        params: SignAndSendTransactionParams
+    ) => Promise<SignAndSendTransactionResponse>;
+
+    /* This function is redundant, since it can be acomplished with signAndSendTransaction.
+    Also, there is high level API in near-api-js Account class */
+    sendMoney: (params: SendMoneyParams) => Promise<unknown>;
+
+    /* Can be usefull, but since it's imposible to do it in NEAR Wallet,
+     * I'm not sure that this should be part of the standart.
+     * Even with Sender wallet, dApp is not always open,
+     * so it should not be a part of the lifecycle.
+     */
+    onAccountChanged: (callback: AccountChangedCallback) => void;
```

Comment:
The [`on(eventName, callback)`](https://docs.senderwallet.io/api-reference/near-provider-api#events) interface will be supported, which mostly comes from the design of MetaMask and can be useful when user switches between their accounts, or any other event happens. This doesn't make sense in web wallet though.